public/cmodel.h

```diff
@@ -59,70 +68,55 @@ struct csurface_t
 //-----------------------------------------------------------------------------
 struct Ray_t
 {
-	VectorAligned  m_Start;	// starting point, centered within the extents
-	VectorAligned  m_Delta;	// direction + length of the ray
-	VectorAligned  m_StartOffset;	// Add this to m_Start to get the actual ray start
-	VectorAligned  m_Extents;	// Describes an axis aligned box extruded along a ray
-	const matrix3x4_t *m_pWorldAxisTransform;
-	bool	m_IsRay;	// are the extents zero?
-	bool	m_IsSwept;	// is delta != 0?
-
-	Ray_t() : m_pWorldAxisTransform( NULL )	{}
-
-	void Init( Vector const& start, Vector const& end )
+	Ray_t() 

```

Comment:
CS2 has a few more constructors which would be nice to have, I guess for every base ray type, altho I haven't seen anything other than Hull_t constructor, like:
```
	Ray_t( const Hull_t &hull )
	{
		Init( hull );
	}

	void Init( const Hull_t &hull )
	{
		if(hull.m_vMin == hull.m_vMax)
		{
			m_eType = RAY_TYPE_LINE;

			VectorCopy( hull.m_vMin, m_Line.m_vStartOffset );
		}
		else
		{
			m_eType = RAY_TYPE_HULL;

			VectorCopy( hull.m_vMin, m_Hull.m_vMin );
			VectorCopy( hull.m_vMax, m_Hull.m_vMax );
		}
	}
```